# <b>Seventh Stage: Introduction of HydABC Complex on iCH360 Model</b>

## 7.1 2-oxoacid:ferredoxin oxidoreductases (OFORs) as $Fd_{red}$ Generator

<div style='text-align: justify'>For $H_2$ production by the HydABC complex, both $Fd_{red}$ and $NADH$ must act as electron donors for the bifurcating mechanism to proceed (<a href="https://doi.org/10.1128/jb.01582-08">2009</a>). Therefore, in addition to $NADH$, $Fd_{red}$ must also be generated by <i>E. coli</i> for the reaction to occur. In this context, several considerations can be made:</div>
<ol>
    <li>In <i>Thermotoga maritima</i>, $Fd_{red}$ is naturally produced by the activity of pyruvate:ferredoxin oxidoreductase (PFOR), an enzyme belonging to the 2-oxoacid:ferredoxin oxidoreductase (OFOR) superfamily (<a href="https://doi.org/10.1016/j.str.2024.08.011">2024</a>, <a href="https://pubs.acs.org/doi/10.1021/bi00170a019">1994</a>). PFOR catalyzes the oxidation of pyruvate coupled to the reduction of $Fd_{ox}$, resulting in the formation of $acetil$-$CoA$, $CO_2$ and $Fd_{red}$ (Reaction 1) (<a href="https://doi.org/10.1021/cr020423e">2003</a>). Additionally, due to the reversibility of this reaction, PFOR can also be referred to as pyruvate synthase (<a href="https://doi.org/10.1021/cr020423e">2003</a>). </li>
</ol>

<div style='text-align: center'>$CoA + pyruvate + Fd_{ox}$ &harr; $acetil$-$CoA + CO_2 + Fd_{red}$ $  $ $(Reaction 1)$</div>

<ol start="2">
    <li>With regard to <i>E. coli</i> metabolism, PFOR mediates the oxidation of pyruvate using both $Fd$ and $Flavodoxin$ as electron carriers. However, as demonstrated by <i>in vitro</i> experiments conducted by Shimizu et al. (<a href="https://doi.org/10.1016/j.ijhydene.2025.04.353">2025</a>), the heterologous expression of a HydABC homolog from <i>Desulfobulbus propionicus</i> in <i>E. coli</i> significantly improves $H_2$ production, particularly when the $Fd$ genes from <i>D. propionicus</i> are also co-expressed. Therefore, in this analysis, although protein affinity cannot be explicitly accounted for, it is assumed that $Fd$ genes from <i>T. maritima</i> are also expressed in <i>E. coli</i> to maximize HydABC activity. It is also important to note that $flavodoxin$ will not be considered in the subsequent analysis, and thus, potential competition between electron carriers will not be evaluated. Finally, in <i>E. coli</i>, PFOR activity is associated with the expression of the ydbK gene, as represented by the corresponding reaction included in the model.</li>
</ol>

<div style='text-align: justify'>Nevertheless, since PDH is an important enzyme for improving $NADH$ yield under anaerobic conditions in <i>E. coli</i> (see Part 6), PFOR may potentially compete with it for pyruvate as a substrate. For this reason, alternative candidates for $Fd_{red}$ generation were considered, such as 2-oxoglutarate:ferredoxin oxidoreductase (OGOR), another member of the OFOR family, and L-glutamate:ferredoxin oxidoreductase (Fd-GOGAT), a key enzyme in nitrogen assimilation. Given the differences in HydABC activity depending on the origin of $Fd$, the expression of a second type of $Fd$-dependent oxidoreductase, such as Fd-GOGAT, could potentially impair the activity of one or both enzymes. In contrast, the similarity between OGOR and PFOR suggests that no major issues related to $Fd$ affinity would be expected. Additionally, OGOR is well aligned with the CS overexpression scenario, as it allows TCA cycle flux to proceed beyond isocitrate dehydrogenase by catalyzing the conversion of 2-oxoglutarate into succinyl-CoA (Reaction 2) (<a href="https://doi.org/10.1128/jb.184.14.3975-3983.2002">2002</a>; <a href="https://doi.org/10.1128/jb.178.20.5890-5896.1996">1996</a>; <a href="https://doi.org/10.1016/S0076-6879(01)31053-4">2001</a>). Therefore, OGOR will also be evaluated as an alternative source of $Fd_{red}$ through heterologous expression, since <i>E. coli</i> does not naturally possess this enzyme in its wild-type metabolism (EcoCyc).</div><br>

<div style='text-align: center'>$2$-$oxaloacetate + CoA + Fd_{ox}$ &harr; $succynil$-$CoA + CO_2 + H^+ + Fd_{red}$ $  $ $(Reaction$ $ 2)$</div><br>

<div style='text-align: justify'>All reactions implemented here were derived from the <a href="http://bigg.ucsd.edu/">BiGG database</a>. For PFOR, the reaction corresponds to a modified version of <a href="http://bigg.ucsd.edu/universal/reactions/PFOR">pyruvate:flavodoxin oxidoreductase</a> (also named PFOR), in which $Fd$ replaces $flavodoxin$ as the electron carrier. Regarding OGOR, the reaction was imported from the <a href="http://bigg.ucsd.edu/models/iLJ478">iLJ478 model</a> of <i>T. maritima</i>, represented by the identifier <a href="http://bigg.ucsd.edu/universal/reactions/OOR2r">OOR2r</a>. Finally, HydABC (Reaction 3) was introduced separately with each enzyme, also derived from the iLJ478 model and represented by the identifier <a href="http://bigg.ucsd.edu/universal/reactions/HYDFDNr">HYDFDN</a>, resulting in the construction of two distinct models: </div>
  
<ol>
    <li><b>EC_iCH360_H2_PFOR:</b> A model containing the heterologous expression of HydABC and the activation of PFOR.</li>
    <li><b>EC_iCH360_H2_OGOR:</b> A model containing the heterologous expression of both HydABC and OGOR from <i>T. maritima.</i></li>
</ol>

<div style='text-align: center'>$3H^+ + NADH + Fd_{red}$ &harr; $2H_2 + NAD^+ + Fd_{ox}$ $  $ $(Reaction$ $ 3)$</div><br>

<div style='text-align: justify'>Both models were subsequently used for further analysis.</div>

## 7.2 Enzyme Pool Calculation

<div style='text-align: justify'>As described in previous sections, the EC_iCH360 model operates with enzyme constraints, represented by a pseudometabolite referred to as the enzyme pool (<a href="https://doi.org/10.1371/journal.pcbi.1013564">Corrao et al., 2025</a>; <a href="https://doi.org/10.1186/s12859-019-3329-9">Bekiaris and Klamt, 2020</a>). In general, this pseudometabolite represents the cost that each reaction must “pay” to carry flux. Accordingly, for each reaction in the model, in both possible directions, an enzyme cost is added as a substrate and calculated as follows:</div>

<div style='text-align: center'>$pool_i = \frac{M_i}{\sigma*k_{cat,i}}$</div><br>

<div style='text-align: justify'>$pool_i$ represents the cost of reaction i; $k_{cat}$ is its turnover number $(h^{-1})$; $M_i$ is the molecular weight of the enzyme (kDA); and $\sigma$ is a unitless, condition-specific scaling factor. (<a href="https://doi.org/10.1371/journal.pcbi.1013564">Corrao et al., 2025</a>).</div><br>

<div style='text-align: justify'>Additionally, a reaction named 'enzyme_pool_supply' is introduced to define the upper bound of enzyme availability that can be allocated across reactions (<a href="https://doi.org/10.1371/journal.pcbi.1013564">Corrao et al., 2025</a>). In this way, the model is constrained not only by reaction flux bounds but also by the total enzyme capacity available for catalysis.</div><br>

<div style='text-align: justify'>In principle, since three new reactions are being added to the model, and all of them are reversible, six $pool$ values must be calculated (one for each reaction direction). However, given the flexibility in the $\sigma$ value that can be adopted, additional considerations must be made. As the $\sigma$ value is indicative of an enzyme's operational capacity, changes in its value could modify model behavior and, consequently, affect $H_2$ production by HydABC. Thus, here, the $\sigma$ value will be set to both 0.25 and 1, allowing observation of how $H_2$ production is affected when all added enzymes operate at a quarter of their maximal capacity and at full capacity, respectively. Hence, to better understand these changes in the newly generated models, we will expand the two models into four: two containing $pool$ values calculated with $\sigma$ equal to 0.25, and two containing $pool$ values calculated with $\sigma$ equal to 1. All implications of such modifications in the $\sigma$ value will be further analyzed in Sections 8, 9, and 10. With this in mind, the calculations and references for these values are presented below:</div>

<div style='text-align: justify'><b>HydABC from <i>T. maritima</i>:</b></div><br>

$M_i$ and $k_{cat}$ values:
<ul>
    <li>$M_i = 330,62$ $kDa$ (<a href="https://www.rcsb.org/structure/7P91">PDB</a>);</li>
    <li>$k_{cat}$ $H_2$ production (fw): $\frac{24+40}{2} * 3600 = 115.200$ $(h^{-1})$ (<a href="https://doi.org/10.1016/S0005-2728(99)00062-6">1999</a>)</li>
    <li>$k_{cat}$ $H_2$ consumption (bw): $\frac{120+190}{2} * 3600 = 558.000$ $(h^{-1})$ (<a href="https://doi.org/10.1016/S0005-2728(99)00062-6">1999</a>)</li>
</ul>

<b>$pool$ values with $\sigma$ set to 0,25:</b>
<ul>
    <li>$pool$ $H_2$ production (fw): $\frac{330,62}{0,25*115.200} = 0,011479861$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $H_2$ consumption (bw): $\frac{330,62}{0,25*558.000} = 0,002370036$ $(g*h*mmol^{-1})$</li>
</ul>


<b>$pool$ values with $\sigma$ set to 1:</b>
<ul>
    <li>$pool$ $H_2$ production (fw): $\frac{330,62}{1*115.200} = 0,0028699653$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $H_2$ consumption (bw): $\frac{330,62}{1*558.000} = 0,000592509$ $(g*h*mmol^{-1})$</li>
</ul>

<div style='text-align: justify'>Note that, for HydABC, $H_2$ production is more “penalized” than $H_2$ consumption.</div><br>

<div style='text-align: justify'><b>PFOR</b>: Unfortunately, specific kinetic parameters for native <i>E. coli</i> PFOR, encoded by the ydbK gene, could not be identified (<a href="https://www.brenda-enzymes.info/enzyme.php?ecno=1.2.8.1">EC. 1.2.8.1</a>). Additionally, no $k_{cat}$ values were found for <i>T. maritima</i> PFOR. Therefore, kinetic parameters from <i>Citrobacter sp.</i>, available in the <a href="https://www.brenda-enzymes.org/enzyme.php?ecno=1.2.7.1">BRENDA database</a>, were used as a proxy. This choice was based on the similarity between the ydbK-encoded protein and PFOR enzymes from <i>Citrobacter sp.</i>, as indicated by <a href="https://www.uniprot.org/uniprotkb/P52647/entry">UniProt</a>. It is important to note, however, that as demonstrated by Shimizu et al. (<a href="https://doi.org/10.1016/j.ijhydene.2025.04.353">2025</a>), endogenous <i>E. coli</i> PFOR is capable of mediating electron transfer to ferredoxin expressed from <i>Desulfobulbus propionicus</i>, which is subsequently utilized by the HydABC complex.</div>

$M_i$ and $k_{cat}$ values:
<ul>
    <li>$M_i = 128,725189$ $kDa$ (<a href="https://doi.org/10.1038/nbt.3418">2016</a>)</li>
    <li>$k_{cat}$ $pyruvate$ consumption (fw): $94 * 3600 = 338.400$ $(h^{-1})$ (<a href="http://dx.doi.org/10.1016/j.biortech.2016.12.051">2017</a>)</li>
    <li>$k_{cat}$ $pyruvate$ production (bw): Not available</li>
</ul>

<b>$pool$ values with $\sigma$ set to 0,5:</b>
<ul>
    <li>$pool$ $pyruvate$ consumption (fw): $\frac{128,725189}{0,5*338.400} = 0,000760787$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $pyruvate$ production (bw): Not available</li>
</ul>

<b>$pool$ values with $\sigma$ set to 1:</b>
<ul>
    <li>$pool$ $pyruvate$ consumption (fw): $\frac{128,725189}{1*338.400} = 0,000380394$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $pyruvate$ production (bw): Not available</li>
</ul><br>

<div style='text-align: justify'><b>PS: For PFOR and OGOR, $\sigma$ value will be set as 0.5. Regarding PFOR, we choose to set $\sigma$ to 0.5 because the enzyme is native of the bacterium, so, we expect that it should have more efficiency on its activity. Additionally, given OGOR similarities with PFOR and its lower complexity compared to HydABC, its $\sigma$ value will also be set to 0.5, as show belows.</b></div><br>

<div style='text-align: justify'><b>OGOR: </b>Similarly to PFOR, specific kinetic parameters for <i>T. maritima</i> OGOR could not be identified. In fact, the presence of OGOR in <i>T. maritima</i> is only indicated through its inclusion in the iLJ478 model. Therefore, kinetic parameters from <i>Magnetococcus marinus</i>, as reported in the <a href="https://www.brenda-enzymes.org/enzyme.php?ecno=1.2.7.3">BRENDA database</a>, were used as a proxy.</div>

$M_i$ and $k_{cat}$ values:
<ul>
    <li>$M_i = 191,12$ $kDa$ (<a href="https://www.rcsb.org/structure/6N2O">PDB</a>)</li>
    <li>$k_{cat}$ $2$-$oxoglutarate$ consumption (fw): $10,8 * 3600 = 38.880$ $(h^{-1})$ (<a href="https://doi.org/10.1016/j.joule.2018.12.006">2019</a>)</li>
    <li>$k_{cat}$ $succinyl$-$CoA$ consumption (bw): $0,45 * 3600 = 1620$ $(h^{-1})$ (<a href="https://doi.org/10.1016/j.joule.2018.12.006">2019</a>)</li>
</ul>

<b>$pool$ values with $\sigma$ set to 0,5:</b>
<ul>
    <li>$pool$ $2$-$oxoglutarate$ consumption (fw): $\frac{191,12}{0,5*38.880} = 0,009831276$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $succinyl$-$CoA$ consumption (bw): $\frac{191,12}{0,5*1620} = 0,235950617$ $(g*h*mmol^{-1})$</li>
</ul>

<b>$pool$ values with $\sigma$ set to 1:</b>
<ul>
    <li>$pool$ $2$-$oxoglutarate$ consumption (fw): $\frac{191,12}{1*38.880} = 0,00491563786$ $(g*h*mmol^{-1})$</li>
    <li>$pool$ $succinyl$-$CoA$ consumption (bw): $\frac{191,12}{1*1620} = 0,11797530864$ $(g*h*mmol^{-1})$</li>
</ul>

<div style='text-align: justify'>For OGOR activity, 2-oxoglutarate consumption is considerably less penalized by the $pool$ value. It is also worth noting that, in contrast to HydABC, both OGOR and PFOR have been experimentally characterized at room temperature, and therefore do not rely on high temperatures to achieve maximum activity, as observed for HydABC. However, homologs of HydABC from non-extremophilic organisms have also been shown to support high $H_2$ yields when heterologously expressed in <i>E. coli</i> (<a href="https://doi.org/10.1016/j.ijhydene.2025.04.353">2025</a>). Additionally, the ability of OGOR to utilize <i>T. maritima</i> (or organisms with homologs HydABC, as <i>D.popionicus</i>) $Fd$ has not been experimentally validated. Therefore, <i>in vitro</i> experiments are required to confirm its capacity for $Fd$ utilization. For the purposes of this analysis, $Fd$ will be assumed to originate from <i>T. maritima</i>, due to its role in HydABC activity. Furthermore, OGOR and PFOR will be assumed to be capable of utilizing this $Fd$. </div><br>

<div style='text-align: justify'>Now, with the $pool$ values calculated, the construction of the models can proceed.</div>

## 7.3 HydABC Expression in the Model

In [1]:
# Model loading and manipulation
import cobra
from cobra.io import load_json_model, save_json_model

In [4]:
# model loading
# model = load_json_model('../Models/EC_iCH360.json')

In [3]:
model

Name,EC_iCH360
Memory address,7f49c0ecf700
Number of metabolites,305
Number of reactions,505
Number of genes,356
Number of groups,0
Objective expression,1.0*Biomass_fw - 1.0*Biomass_fw_reverse_bf19d
Compartments,"cytosol, extracellular space, periplasm"


In [3]:
# For practical purposes, the anaerobic simulated environment will be stored within the models
model_anaero = model.copy()
model_anaero.reactions.get_by_id('EX_o2_e_bw').bounds = (0,0)

Read LP format model from file /tmp/tmpmz60gx75.lp
Reading time = 0.00 seconds
: 305 rows, 1010 columns, 4816 nonzeros


In [4]:
# Preparation of HydABC model

# model with sigma equal to 1.0
model_anaero_H2_1 = model_anaero.copy()

# model with sigma equal to 0.25
model_anaero_H2_25 = model_anaero.copy()

Read LP format model from file /tmp/tmpn8ncg6ks.lp
Reading time = 0.00 seconds
: 305 rows, 1010 columns, 4816 nonzeros
Read LP format model from file /tmp/tmpzvg0_9u5.lp
Reading time = 0.00 seconds
: 305 rows, 1010 columns, 4816 nonzeros


In [5]:
# Adding HydABC reactions to model_anaero_H2_1
model_anaero_H2_1.add_metabolites([cobra.Metabolite('fdxrd_c',compartment='c'),
                                   cobra.Metabolite('fdxox_c',compartment='c'),   
                                   cobra.Metabolite('h2_c',compartment='c'),
                                   cobra.Metabolite('h2_e',compartment='e'),
                                   cobra.Metabolite('h2_p',compartment='p')
                                  ])

HYDFDN_fw_1 = cobra.Reaction('HYDFDN_fw')
HYDFDN_bw_1 = cobra.Reaction('HYDFDN_bw')
EX_h2_e_fw_1 = cobra.Reaction('EX_h2_e_fw')
H2TPP_fw_1 = cobra.Reaction('H2TPP_fw')
H2tex_fw_1 = cobra.Reaction('H2tex_fw')

model_anaero_H2_1.add_reactions([HYDFDN_fw_1,
                                 HYDFDN_bw_1,
                                 EX_h2_e_fw_1,
                                 H2TPP_fw_1,
                                 H2tex_fw_1])

# Define reaction equations
HYDFDN_fw_1.reaction = '3.0 h_c + 1.0 nadh_c + 1.0 fdxrd_c + 0.0028699653 enzyme_pool -> 2.0 h2_c + 1.0 nad_c + 1.0 fdxox_c'
HYDFDN_bw_1.reaction = '2.0 h2_c + 1.0 nad_c + 1.0 fdxox_c + 0.000592509 enzyme_pool -> 3.0 h_c + 1.0 nadh_c + 1.0 fdxrd_c'
H2TPP_fw_1.reaction = '1 h2_c -> 1 h2_p'
H2tex_fw_1.reaction='1 h2_p -> 1 h2_e' 
EX_h2_e_fw_1.reaction = '1 h2_e ->'

In [6]:
# Adding HydABC reactions to model_anaero_H2_25
model_anaero_H2_25.add_metabolites([cobra.Metabolite('fdxrd_c',compartment='c'),
                                    cobra.Metabolite('fdxox_c',compartment='c'),   
                                    cobra.Metabolite('h2_c',compartment='c'),
                                    cobra.Metabolite('h2_e',compartment='e'),
                                    cobra.Metabolite('h2_p',compartment='p')
                                   ])

HYDFDN_fw_25 = cobra.Reaction('HYDFDN_fw')
HYDFDN_bw_25 = cobra.Reaction('HYDFDN_bw')
EX_h2_e_fw_25 = cobra.Reaction('EX_h2_e_fw')
H2TPP_fw_25 = cobra.Reaction('H2TPP_fw')
H2tex_fw_25 = cobra.Reaction('H2tex_fw')

model_anaero_H2_25.add_reactions([HYDFDN_fw_25,
                                 HYDFDN_bw_25,
                                 EX_h2_e_fw_25,
                                 H2TPP_fw_25,
                                 H2tex_fw_25])

# Define reaction equations
HYDFDN_fw_25.reaction = '3.0 h_c + 1.0 nadh_c + 1.0 fdxrd_c + 0.011479861 enzyme_pool -> 2.0 h2_c + 1.0 nad_c + 1.0 fdxox_c'
HYDFDN_bw_25.reaction = '2.0 h2_c + 1.0 nad_c + 1.0 fdxox_c + 0.002370036 enzyme_pool -> 3.0 h_c + 1.0 nadh_c + 1.0 fdxrd_c'
H2TPP_fw_25.reaction = '1 h2_c -> 1 h2_p'
H2tex_fw_25.reaction='1 h2_p -> 1 h2_e' 
EX_h2_e_fw_25.reaction = '1 h2_e ->'

In [7]:
# Creation of the four new models with added reactions

# Models with PFOR and HydABC
model_anaero_H2_PFOR_1 = model_anaero_H2_1.copy()
model_anaero_H2_PFOR_25 = model_anaero_H2_25.copy()

# Models with OGOR and HydABC
model_anaero_H2_OGOR_1 = model_anaero_H2_1.copy()
model_anaero_H2_OGOR_25 = model_anaero_H2_25.copy()

Read LP format model from file /tmp/tmpjea02wbz.lp
Reading time = 0.00 seconds
: 310 rows, 1020 columns, 4854 nonzeros
Read LP format model from file /tmp/tmpibbz5cvb.lp
Reading time = 0.00 seconds
: 310 rows, 1020 columns, 4854 nonzeros
Read LP format model from file /tmp/tmple0bcao8.lp
Reading time = 0.00 seconds
: 310 rows, 1020 columns, 4854 nonzeros
Read LP format model from file /tmp/tmpsizmy1g_.lp
Reading time = 0.00 seconds
: 310 rows, 1020 columns, 4854 nonzeros


<div style='text-align: justify'>However, since the model does not include any reaction that produces $Fd_{red}$ for $H_2$ production via HydABC, PFOR and OGOR must be added to the models, as follows:</div>

## 7.4 HydABC and PFOR

In [8]:
# Adding PFOR reaction to model_anaero_H2_PFOR_1
# Since no data are available for the pool value of PFOR_bw, this reaction was excluded, as the absence of an enzyme cost could artificially favor its flux. 

PFOR_fw_1 = cobra.Reaction('PFOR_fw')
model_anaero_H2_PFOR_1.add_reactions([PFOR_fw_1])

# Define reaction equations
PFOR_fw_1.reaction = '1.0 coa_c + 1.0 pyr_c + 1.0 h_c + 1.0 fdxox_c + 0.000380394 enzyme_pool -> 1.0 accoa_c + 1.0 co2_c + 1.0 fdxrd_c'

In [9]:
# Adding PFOR reaction to model_anaero_H2_PFOR_05 

PFOR_fw_05 = cobra.Reaction('PFOR_fw')
model_anaero_H2_PFOR_25.add_reactions([PFOR_fw_05])

# Define reaction equations
PFOR_fw_05.reaction = '1.0 coa_c + 1.0 pyr_c + 1.0 h_c + 1.0 fdxox_c + 0.000760787 enzyme_pool -> 1.0 accoa_c + 1.0 co2_c + 1.0 fdxrd_c'

<div style='text-align: justify'><b>Considerations About Enzyme abundance</b>

In [10]:
model.reactions.get_by_id('enzyme_pool_supply')

Reaction identifier,enzyme_pool_supply
Name,enzyme pool pseudoreaction
Memory address,0x7f81e1ce1030
Stoichiometry,--> enzyme_pool --> enzyme pool pseudometabolite
GPR,
Lower bound,0.0
Upper bound,0.281838974245756


<div style='text-align: justify'> In all optimizations performed on EC_iCH360, enzyme_pool_supply is considered the maximum enzyme quantity that can be allocated to the specified reactions. In this way, through the pseudometabolite $pool$, reactions are also constrained by the amount of enzyme required to sustain their activity. The calculation and explanation of the pseudometabolite and enzyme pool supply can be visualized in detail in the work of <a href="https://doi.org/10.1371/journal.pcbi.1013564">Corrao et al., (2025)</a>.</div><br>

<div style='text-align: justify'> Hereby, since we are adding three new reactions to the model, we must rethink the enzyme pool for each newly introduced enzyme. As HydABC and OGOR are heterologous enzymes in <i>E. coli</i>, we can assume that they do not affect the enzyme pool supply, and instead compete for enzyme allocation with endogenous enzymes. Therefore, the choice of whether to use these new enzymes will be based not only on their influence on flux distribution but also on their effects on total enzyme consumption. In contrast, with regard to PFOR, additional considerations must be made, since PFOR is not a heterologous enzyme in <i>E. coli</i> metabolism. The addition of PFOR implies that we are expanding EC_iCH360 model to include reactions that were not previously incorporated but have been annotated in <i>in vitro</i> experiments.</div><br>

<div style='text-align: justify'> Hence, models with PFOR must be recalibrated with respect to enzyme pool, in order to include the contribution of PFOR in the reaction 'enzyme_pool_supply'. The calculation of the PFOR pool value is based on experimental data available from <a href="https://doi.org/10.1038/nbt.3418">Schmidt et al., (2016)</a>, and processed by <a href="https://github.com/marco-corrao/iCH360_paper/blob/main/Enzyme_Constraints/data/Schmidt_et_al_2016/parsed_data/Schmidt_et_al_2016_parsed.csv">Corrao et al., (2025)</a>. PFOR enzyme abundance will be calculated based on the median of all enzymatic quantities of the enzyme found under different growth conditions, in grams of enzyme per gram of cell dry weight (g/gDW). Each selected value corresponds to PFOR abundance, in g/gDW, under different growth conditions. PFOR was considered as the product of ydbK gene expression, found on line 1944 of <a href="https://github.com/marco-corrao/iCH360_paper/blob/main/Enzyme_Constraints/data/Schmidt_et_al_2016/parsed_data/Schmidt_et_al_2016_parsed.csv">Corrao et al., (2025)</a> dataset:</a></div>

<ul>
    <li><b>glucose:</b> 4.96224396557859e-05;</li>
    <li><b>acetate:</b> 3.5880840981876e-05;</li>
    <li><b>fumarate:</b> 3.81711074275276e-05;</li>
    <li><b>glycerol:</b> 4.58053289130332e-05;</li>
    <li><b>pyruvate:</b> 3.13003080905727e-05;</li>
    <li><b>fructose:</b> 6.33640383296959e-05;</li>
    <li><b>succinate:</b> 4.96224396557859e-05;</li>
    <li><b>xylose:</b> 3.96979517246287e-05;</li>
    <li><b>glucose chemostat $\mu$ 0.5:</b> 4.73321732101343e-05;</li>
    <li><b>glucose chemostat $\mu$ 0.12:</b> 6.87079933695497e-05;</li>
    <li><b>glucose chemostat $\mu$ 0.20:</b> 5.03858618043365e-05;</li>
    <li><b>glucose chemostat $\mu$ 0.35:</b> 4.50419067644826e-05;</li>
    <li><b>glucose MG1655:</b> 2.9010041644921e-05;</li>
</ul>

<div style='text-align: justify'> With this, the calculated median of PFOR enzyme abundance is: $4.568787935×10^{-5}$ $g*gDW^{-1}$. Now, we must add it to the enzyme_pool_supply upper bound:</div><br>

<div style='text-align: center'> $final\_pool\_supply = 0.281838974245756 + 0.00004568787935 = 0.2818846621$</div><br>

<div style='text-align: justify'> All models with PFOR (<b>EC_iCH360_H2_PFOR_05</b> and <b>EC_iCH360_H2_PFOR_1</b>) must be modified to incorporate the new enzyme_pool_supply value. Models <b>EC_iCH360_H2_OGOR_05</b> and <b>EC_iCH360_H2_OGOR_1</b> will not be modified with respect to enzyme_pool_supply, as discussed previously.</div>

In [11]:
# Changing the enzyme_pool_supply reaction in models with the PFOR reaction to the updated one

# model_anaero_H2_PFOR_1
model_anaero_H2_PFOR_1.reactions.get_by_id('enzyme_pool_supply').upper_bound = 0.2818846621

# model_anaero_H2_PFOR_05
model_anaero_H2_PFOR_25.reactions.get_by_id('enzyme_pool_supply').upper_bound = 0.2818846621

In [12]:
# Analysis of biomass production and fluxes of introduced enzymes

# model_anaero_H2_PFOR_1
print("model_anaero_H2_PFOR_1 solution:\n")
solution = model_anaero_H2_PFOR_1.optimize()
display(solution)
print('')

# model_anaero_H2_PFOR_05
print("model_anaero_H2_PFOR_25 solution:\n")
solution = model_anaero_H2_PFOR_25.optimize()
display(solution)

model_anaero_H2_PFOR_1 solution:



,fluxes,reduced_costs
NDPK5_fw,0.003339,-7.589415e-18
SHK3Dr_fw,0.046967,1.585646e-18
NDPK6_fw,0.003234,-1.665335e-16
NDPK8_fw,0.003234,8.673617e-19
DHORTS_fw,0.000000,-8.674663e-03
...,...,...
HYDFDN_bw,0.000000,-3.375384e-03
EX_h2_e_fw,15.762880,0.000000e+00
H2TPP_fw,15.762880,0.000000e+00
H2tex_fw,15.762880,0.000000e+00



model_anaero_H2_PFOR_25 solution:



,fluxes,reduced_costs
NDPK5_fw,0.002864,2.450297e-17
SHK3Dr_fw,0.040289,-5.014435e-19
NDPK6_fw,0.002774,1.804112e-16
NDPK8_fw,0.002774,-7.589415e-19
DHORTS_fw,0.000000,-7.838896e-03
...,...,...
HYDFDN_bw,0.000000,-1.220072e-02
EX_h2_e_fw,0.921608,0.000000e+00
H2TPP_fw,0.921608,0.000000e+00
H2tex_fw,0.921608,0.000000e+00


<div style='text-align: justify'>From the demonstrated fluxes and objective values, we can see that flux through HydABC is not blocked and may additionally improve biomass production when compared with the wild-type model. It is also noted that the $\sigma$ value has a quantitative effect on model simulations. Nonetheless, at least for HydABC and $H_2$ export reactions, the general behavior remains the same between the models, in which both the complex and PFOR are activated to improve biomass production. Thus, to account for quantitative differences and potential qualitative modifications in further analyses, in the next chapters all simulations will be performed for both models: model_anaero_H2_PFOR_1 and model_anaero_H2_PFOR_05.</div>

## 7.5 HydABC and OGOR

In [13]:
# Adding OGOR reaction to model_anaero_H2_OGOR_1
OGOR_fw_1 = cobra.Reaction('OGOR_fw')
OGOR_bw_1 = cobra.Reaction('OGOR_bw')
model_anaero_H2_OGOR_1.add_reactions([OGOR_fw_1,OGOR_bw_1])

# Define reaction equations
OGOR_fw_1.reaction = '1.0 akg_c + 1.0 coa_c + 1.0 fdxox_c + 0.00491563786 enzyme_pool -> 1.0 succoa_c + 1.0 co2_c + 1.0 fdxrd_c + 1.0 h_c'
OGOR_bw_1.reaction = '1.0 succoa_c + 1.0 co2_c + 1.0 fdxrd_c + 1.0 h_c + 0.11797530864 enzyme_pool -> 1.0 akg_c + 1.0 coa_c + 1.0 fdxox_c'

In [14]:
# Adding OGOR reaction to model_anaero_H2_OGOR_05
OGOR_fw_05 = cobra.Reaction('OGOR_fw')
OGOR_bw_05 = cobra.Reaction('OGOR_bw')
model_anaero_H2_OGOR_25.add_reactions([OGOR_fw_05,OGOR_bw_05])

# Define reaction equations
OGOR_fw_05.reaction = '1.0 akg_c + 1.0 coa_c + 1.0 fdxox_c + 0.009831276 enzyme_pool -> 1.0 succoa_c + 1.0 co2_c + 1.0 fdxrd_c + 1.0 h_c'
OGOR_bw_05.reaction = '1.0 succoa_c + 1.0 co2_c + 1.0 fdxrd_c + 1.0 h_c + 0.235950617 enzyme_pool -> 1.0 akg_c + 1.0 coa_c + 1.0 fdxox_c'

In [17]:
# Analysis of biomass production and fluxes of introduced enzymes

# model_anaero_H2_OGOR_1
print("model_anaero_H2_OGOR_1 solution:\n")
solution_1 = model_anaero_H2_OGOR_1.optimize()
display(solution_1)

# model_anaero_H2_OGOR_05
print("model_anaero_H2_OGOR_25 solution:\n")
solution_25 = model_anaero_H2_OGOR_25.optimize()
display(solution_25)

model_anaero_H2_OGOR_1 solution:



,fluxes,reduced_costs
NDPK5_fw,0.002788,1.084202e-17
SHK3Dr_fw,0.039217,9.676504e-18
NDPK6_fw,0.002700,-1.387779e-17
NDPK8_fw,0.002700,-1.268517e-17
DHORTS_fw,0.000000,-7.631694e-03
...,...,...
EX_h2_e_fw,0.000000,0.000000e+00
H2TPP_fw,0.000000,-1.800986e-02
H2tex_fw,0.000000,0.000000e+00
OGOR_fw,0.000000,-1.301043e-18


model_anaero_H2_OGOR_25 solution:



,fluxes,reduced_costs
NDPK5_fw,0.002788,1.084202e-17
SHK3Dr_fw,0.039217,9.676504e-18
NDPK6_fw,0.002700,-1.387779e-17
NDPK8_fw,0.002700,-1.268517e-17
DHORTS_fw,0.000000,-7.631694e-03
...,...,...
EX_h2_e_fw,0.000000,0.000000e+00
H2TPP_fw,0.000000,-1.935555e-02
H2tex_fw,0.000000,0.000000e+00
OGOR_fw,0.000000,2.602085e-18


<div style='text-align: justify'>Contrary to what is observed for PFOR/HydABC expression, OGOR expression has no effect on model behavior. As shown, all introduced enzymes carry no flux, and the models behave identically to the wild-type anaerobic model. However, it is important to determine whether the added reactions are inactive due to their low relevance for objective maximization (biomass) or because they were incorrectly implemented and are therefore blocked.</div>

In [18]:
# Let us change the model objective to OGOR to evaluate whether the reaction is blocked

# model_anaero_H2_OGOR_1
with model_anaero_H2_OGOR_1 as m_o:
    m_o.objective = 'OGOR_fw'
    sol = m_o.optimize()
    display(sol)

# model_anaero_H2_OGOR_05
with model_anaero_H2_OGOR_25 as m_o:
    m_o.objective = 'OGOR_fw'
    sol = m_o.optimize()
    display(sol)

,fluxes,reduced_costs
NDPK5_fw,0.000000,4.163336e-17
SHK3Dr_fw,0.000000,-4.206704e-17
NDPK6_fw,0.000000,-1.132427e-14
NDPK8_fw,0.000000,-2.914335e-16
DHORTS_fw,0.000000,-3.071714e-01
...,...,...
EX_h2_e_fw,8.307536,0.000000e+00
H2TPP_fw,8.307536,0.000000e+00
H2tex_fw,8.307536,0.000000e+00
OGOR_fw,4.153768,-3.330669e-16


,fluxes,reduced_costs
NDPK5_fw,0.000000,-1.318390e-16
SHK3Dr_fw,0.000000,-6.071532e-18
NDPK6_fw,0.000000,4.662937e-15
NDPK8_fw,0.000000,2.116363e-16
DHORTS_fw,0.000000,-2.490348e-01
...,...,...
EX_h2_e_fw,6.735217,0.000000e+00
H2TPP_fw,6.735217,0.000000e+00
H2tex_fw,6.735217,0.000000e+00
OGOR_fw,3.367608,-1.387779e-16


<div style='text-align: justify'>By modifying the model objective, it is possible to visualize that neither OGOR nor HydABC is blocked. It is also important to note that OGOR_bw is practically inactivated due to its extremely high enzyme cost, making the OGOR reaction almost irreversible. Lastly, as expected, we can infer that the HydABC complex is strongly tied to the $Fd_{red}$ generation reaction, such that $H_2$ production is inactive if $Fd_{red}$ generation is not feasible for maximization of the model objective.</div>

## 7.6 Considerations

### 7.6.1 PFOR/HydABC

<div style='text-align: justify'>Initially, it can be observed that, in EC_iCH360_PFOR_1 model, the growth rate changes significantly after the expression of PFOR and HydABC, reaching a biomass yield of 0.124 $h^{-1}$ (for comparison, the wild-type biomass yield under anaerobic conditions is 0.1032 $h^{-1}$). In this way, when at maximal operational capacity for all introduced enzymes, biomass production is considerably affected by PFOR and HydABC expression. Additionally, all introduced reactions operate at considerable fluxes, indicating that the pathway is, at least stoichiometrically, suitable not only for improving biomass production but also for NADH consumption, since HydABC flux accounts for NADH consumption.</div>

### 7.6.2 OGOR/HydABC

<div style='text-align: justify'>In contrast to PFOR, OGOR is not induced to operate on both generated models. This results suggests that the generation of 2-oxoglutarate is not favorable for biomass maximization, leading the model to behave similarly to the wild type anaerobic condition. However, when the model objective is set to OGOR_fw, the reaction is shown to be unblocked and, when active, is capable of generating $Fd_{red}$ for HydABC utilization. It is expected that OGOR may be more suitable in scenarios involving PDH and CS overexpression, due to the potential coupling between CS activity and OGOR flux, with OGOR supplying $Fd_{red}$ to HydABC. Further analysis of model behavior under these modified conditions will be presented in Section 8. Here, the focus is limited to validating the calculation of $pool$ values and confirming the correct implementation of the added enzymes.</div><br>

<div style='text-align: justify'><b>The models can now be saved for further analyses and modifications.</div>

In [19]:
# saving the models

# model_anaero_H2_PFOR_25 and model_anaero_H2_PFOR_1
save_json_model(model_anaero_H2_PFOR_1,'../Models/EC_iCH360_H2_PFOR_1.json')
save_json_model(model_anaero_H2_PFOR_25,'../Models/EC_iCH360_H2_PFOR_25.json')

# model_anaero_H2_OGOR_25 and model_anaero_H2_OGOR_1
save_json_model(model_anaero_H2_OGOR_1,'../Models/EC_iCH360_H2_OGOR_1.json')
save_json_model(model_anaero_H2_OGOR_25,'../Models/EC_iCH360_H2_OGOR_25.json')

# <b>End of Part 7!</b>